cmd/suite_test.go

```diff
@@ -0,0 +1,155 @@
+                        "totalCount": 1,
+                        "nodes": [
+                            {
+                                "commit": {
+                                    "committedDate": "2022-03-21T15:09:52Z"
+                                }
+                            }
+                        ]
+                    },
+                    "timelineItems": {
+                        "totalCount": 1,
+                        "nodes": [
+                            {
+                                "createdAt": "2022-03-15T03:46:20Z"
+                            }
+                        ]
+                    }
+                },
+                {
+                    "author": {
+                        "login": "Batman"
+                    },
+                    "additions": 12,
+                    "deletions": 6,
+                    "number": 5340,
+                    "createdAt": "2022-03-22T15:11:09Z",
+                    "changedFiles": 2,
+                    "isDraft": false,
+                    "mergedAt": "2022-03-22T16:22:05Z",
+                    "participants": {
+                        "totalCount": 3
+                    },
+                    "comments": {
+                        "totalCount": 0
+                    },
+                    "reviews": {
+                        "nodes": [
+                            {
+                                "author": {
+                                    "login": "Joker"
+                                },
+                                "createdAt": "2022-03-22T15:12:52Z",
+                                "state": "COMMENTED"
+                            },
+                            {
+                                "author": {
+                                    "login": "Joker"
+                                },
+                                "createdAt": "2022-03-23T15:12:52Z",
+                                "state": "APPROVED"
+                            }
+                        ]
+                    },
+                    "commits": {
+                        "totalCount": 1,
+                        "nodes": [
+                            {
+                                "commit": {
+                                    "committedDate": "2022-03-22T15:09:52Z"
+                                }
+                            }
+                        ]
+                    },
+                    "timelineItems": {
+                        "totalCount": 1,
+                        "nodes": [
+                            {
+                                "createdAt": "2022-03-16T03:46:20Z"
+                            }
+                        ]
+                    }
+                }
+            ]
+        }
+    }
+}`
+)
+
+type GQLRequest struct {
+	Variables struct {
+		Query string
+	}
+}
+
+func gqlSearchQueryMatcher(owner, repo, start, end string) func(req *http.Request, ereq *gock.Request) (bool, error) {
```

Comment:
Now, `gqlSearchQueryMatcher` accepts owner, repo, start, and end arguments such that we can more granularly configure matching on a per-test basis.